pkg/apis/common/v1/types.go

```diff
@@ -22,6 +22,7 @@ import (
 // +k8s:deepcopy-gen=true
 // JobStatus represents the current observed state of the training Job.
 type JobStatus struct {
+	// +listType=set
```

Comment:
this should be a map, with listMapKey=type

Reply:
I'm. afraid there can be conflicts if it's map with `listMapKey=type`. For example, there can be multiple items with the same JobCondition like `Running`.  I think there can't be duplicated items so maybe set is suitable?

Reply:
Why would there be multiple conditions for the same type?

Reply:
Condition is a list and it's open to user to set so it may happen though it's not recommended. But you are right I checked code for JobStatus it's impossible this way. I can modify it then.

Reply:
The JobStatus is actually not up to the user, it's entirely controlled by the operators. And if our operators put the same condition more than once, it would be a bug. But generally, lists could act as maps if we decide to.

Reply:
OK, send another PR https://github.com/kubeflow/common/pull/174 to address this.